tfjs-backend-webgpu/src/kernels/argminmax_webgpu.ts

```diff
@@ -164,31 +122,51 @@ export class ArgMinMaxProgram implements WebGPUProgram {
         return coordInfo[0] + coordInfo[1] * index;
       }
 
-      ${getNonFlatDispatchLayoutMainHeaderString()} {
-        let coordInfo = getInputCoordInfo(globalId);
+      ${getMainHeaderAndGlobalIndexString()}
+        let outputIndex = index / i32(workGroupSizeX);
+        let coordInfo = getInputCoordInfo(outputIndex);
+        let Length = ${indexInputShape('uniforms.axis')};
 
-        var bestIndex = 0;
-        var bestValue = f32(x.numbers[getInputIndex(coordInfo, bestIndex)]);
+        var bestIndex = i32(localId.x);
+        var bestValue = uniforms.infinityValue;
 
-        let Length = ${indexInputShape('uniforms.axis')};
-        let WorkPerThread = DIV_CEIL(Length, WorkGroupSize);
-
-        for (var w = 0; w < WorkPerThread; w = w + 1) {
-          let i = i32(globalId.x) * WorkPerThread + w;
-          if (i < Length) {
-            let candidate = f32(x.numbers[getInputIndex(coordInfo, i)]);
-            if (candidate ${
-        this.op} bestValue && !isNanCustom(f32(candidate))) {
+        if (i32(localId.x) < Length && outputIndex < uniforms.size) {
+          let candidate = f32(x.numbers[getInputIndex(coordInfo, bestIndex)]);
+          if (!isNanCustom(candidate)) {
               bestValue = candidate;
-              bestIndex = i;
+          }
+        }
+
+        for (var k = i32(localId.x) + i32(workGroupSizeX); k < Length &&
+            outputIndex < uniforms.size; k = k + i32(workGroupSizeX)) {
+          let candidate = f32(x.numbers[getInputIndex(coordInfo, k)]);
+          if (candidate ${this.op} bestValue && !isNanCustom(candidate)) {
+            bestValue = candidate;
+            bestIndex = k;
+          }
+        }
+        xBestValues[localId.x] = bestValue;
+        xBestIndices[localId.x] = bestIndex;
+        workgroupBarrier();
+
+        var reduceSize = min(u32(Length), workGroupSizeX);
+        for (var currentSize = reduceSize / 2u; reduceSize > 1u;
+            currentSize = reduceSize / 2u) {
+          let interval = DIV_CEIL(reduceSize, 2u);
```

Comment:
Because they are not the same value if `reduceSize` is odd number, take `reduceSize = 5` for an example, `currentSize = 2` and `interval = 3` here. their indices are [0, 1, 2, 3, 4], we only compare 0 <-> 3 and 1 <-> 4.